### ISSUE TITLE
Accessibility of the Captions Settings dialog

### ISSUE THREAD

**Issue by OwenEdwards** [videojs/video.js#2746]
The Captions Settings dialog has some accessibility issues:
- [x] Field labels and fields are not explicitly associated
- [ ] Keyboard focus does not move into the dialog when it is opened
- [ ] Keyboard focus is not trapped inside the dialog while it is open
- [ ] Keyboard focus does not return to the control which opened the dialog when it is closed
- [ ] The extent (top and bottom) of the dialog is not indicated to screen readers
- [ ] The dialog cannot be closed with the Esc key
- [x] The meaning of '---' in the select fields is not clear
- [x] The control to close the dialog is labeled "Done" rather than "Close"
- [ ] The purpose of the "Defaults" button may not be obvious, and its effect may not be apparent to screen reader users

Some of these are very simple fixes, and some are lower priority than others, but all are fixable :-)

**Comment by OwenEdwards** [comment]
Much of this can be adopted from @gdkraus' excellent example https://github.com/gdkraus/accessible-modal-dialog.

**Comment by OwenEdwards** [comment]
@gkatsev, to clarify your points on the #2668 Modal Dialog thread,

1. I understand that, while the Captions Setting dialog is displayed then per FCC the user needs to see the effect of the settings I don't think having the Captions Setting dialog as a modal dialog excludes being able to see the current captions, unless it is implemented as a lightbox. The modal dialog doesn't have to be as big as the player window, does it?

2. Some players use an example of caption text in the settings dialog itself, so the existing captions *could* be obscured.

3. I guess I need to think about what I mean by modal dialog here; I get the point on #2668 that you want users to still be able to interact with the rest of the page, so the settings dialog shouldn't be modal in that sense, but I'm thinking of it more from a keyboard focus perspective, in terms of making sure that keyboard-only users can't accidentally move focus out of the dialog into player controls and then not be clear how to get focus back into the dialog.

Keyboard focus is something we need to think about broadly for accessibility, so let's give 3 some more thought. Can you clarify what you meant wrt 1 (and whether 2 might be an option? One of the issues I see with the Captions Settings dialog is sizing it for different video window sizes, which might make 2 more appealing?).

**Comment by gkatsev** [comment]
1. Yep, the modal dialog could be sized down.
2. It's definitely doable to have a preview section just would require some more refactoring of text tracks and rendering portions.
3. We probably want both a modal and non-modal dialogs with the current incarnation of the settings dialog being a non-modal dialog, though, if 2 is made, it could be a modal dialog.
4. Yeah, we really need to work on our focus... I mean, keyboard focus.

**Comment by gkatsev** [comment]
The dialog focus will be solved by https://github.com/videojs/video.js/pull/4025

**Comment by gkatsev** [comment]
While the focus issues will be solved by #4025, the captions dialog is still currently extending Component.
